### PR TITLE
docs(install): Document how to install and uninstall with Helm

### PR DESCRIPTION
**Pull request by ksubrmnn**
Signed-off-by: Kalya Subramanian <kasubra@microsoft.com>

**Description**:
Documents how to install and uninstall osm with helm cli

**Affected area**:

- New Functionality      [ ]
- Documentation          [X]
- Install                [ ]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Performance            [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
 No

### CODE REVIEW

**Review by nojnhuh — APPROVED** [review]

File: docs/content/docs/install/_index.md [hunk 1]
```diff
@@ -61,7 +87,7 @@ To install OSM on OpenShift:
     ```shell
     osm install --set="OpenServiceMesh.enablePrivilegedInitContainer=true"
     ```
-    - If you have already installed OSM without enabling privileged init containers, set `enable_privileged_init_container` to `true` in the [OSM ConfigMap](../osm_config_map.md) and restart any pods in the mesh.
+    - If you have already installed OSM without enabling privileged init containers, set `enable_privileged_init_container` to `true` in the [OSM ConfigMap](../osm_config_map) and restart any pods in the mesh.
```
Comment (shashankram) [inline docs/content/docs/install/_index.md:90]: I thought we need to use the file extension (`../osm_config_map.md`) and use an alias of `osm_config_map.md` for the configmap doc so it renders fine on the website.

Why is the extension being removed?

Comment (shashankram) [reply docs/content/docs/install/_index.md:90]: I am referring to how internal links are to be referenced: https://github.com/openservicemesh/osm/tree/main/docs#internal-links

File: docs/content/docs/install/_index.md [hunk 2]
```diff
@@ -43,16 +56,29 @@ $ osm install
 OSM installed successfully in namespace [osm-system] with mesh name [osm]
 ```
 
-By default, the control plane components are installed into a Kubernetes Namespace called `osm-system` and the control plane is given a unique identifier attribute `mesh-name` defaulted to `osm`. Both the Namespace and mesh-name can be configured with flags to the `osm install` command. Running `osm install --help` provides details on the various flags that can be configured.
+Run `osm install --help` for more options.
 
-The `mesh-name` is a unique identifier assigned to an osm-controller instance during install to identify and manage a mesh instance.
+### Using the Helm CLI
```
Comment (michelleN) [inline docs/content/docs/install/_index.md:61]: I would add a line here stating that the Helm chart is located at github.com/openservicemesh/osm/charts/osm

File: docs/content/docs/install/_index.md (hunk 1 above)
Comment (shashankram) [reply docs/content/docs/install/_index.md:90]: @ksubrmnn , based on discussions offline, we don't need to have this link navigable on github. My original comment can be ignored.

**Review by shashankram — APPROVED** [review]

**Review by draychev — APPROVED** [review]

File: docs/content/docs/upgrade_guide.md [hunk 3]
```diff
@@ -94,7 +94,7 @@ When upgrading, any custom settings used to install or run OSM may be reverted t
 
 To preserve any changes you've made to the OSM configuration, use the `helm --values` flag. Create a copy of the [values file](https://github.com/openservicemesh/osm/blob/main/charts/osm/values.yaml) (make sure to use the version for the upgraded chart) and change any values you wish to customize. You can omit all other values.
 
-To see which values correspond to the ConfigMap settings, see the [OSM ConfigMap documentation](osm_config_map.md)
+To see which values correspond to the ConfigMap settings, see the [OSM ConfigMap documentation](osm_config_map)
```
[the remark this answers is not in the document]
Comment (shashankram) [reply docs/content/docs/upgrade_guide.md:97]: If using the `.md` extension, make sure the file as an alias so it renders on the website.

**Pull request merged**